docs/content/what-is-a-task-queue.md

```diff
@@ -43,3 +43,15 @@ This implementation offers several benefits:
 
 All Workers listening to a given Task Queue must have identical registrations of Activities and/or Workflows.
 The one exception is during a Server upgrade, where it is okay to have registration temporarily misaligned while the binary rolls out.
+
+<details>
+  <summary>In what order does the Task Queue match Tasks? FIFO? LIFO? Random?</summary>
+  
+  Temporal offers no guarantees as to the order that Tasks are matched (popped) off the Task Queue.
+  There are two kinds of matching channels ("sync match" and "async match") based on where the tasks are located.
+  Tasks that fail to sync match will overflow to the persistence database, and be picked up in a separate channel.
+  Temporal then selects from both channels at random.
```

Comment:
IMHO this is too confusing for everyone who doesn't know about Go channels. 

I would just say that Temporal doesn't guarantee any order and it is impossible to have a task that stays in a queue for a long time if there is a backlog that wasn't drained for that time.